include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -117,6 +118,16 @@ typename MatType::elem_type NSGA2::Optimize(
   {
     population.push_back(arma::randu<MatType>(iterate.n_rows,
         iterate.n_cols) - 0.5 + iterate);
+    
+    // Ensure population is within variable space
+    for(size_t geneIdx = 0; geneIdx < numVariables; geneIdx++)
+    {
+      if(population[i][geneIdx] < lowerBound[geneIdx])
+        population[i][geneIdx] = lowerBound[geneIdx];
+      
+      if(population[i][geneIdx] > upperBound[geneIdx])
+        population[i][geneIdx] = upperBound[geneIdx];
+    }
```

Comment:
```suggestion
    // Ensure population is within variable space.
    for (size_t geneIdx = 0; geneIdx < numVariables; geneIdx++)
    {
      if (population[i][geneIdx] < lowerBound[geneIdx])
        population[i][geneIdx] = lowerBound[geneIdx];
      
      if (population[i][geneIdx] > upperBound[geneIdx])
        population[i][geneIdx] = upperBound[geneIdx];
    }
```

We tend to be picky about the style.